Code/Framework/AzToolsFramework/AzToolsFramework/ComponentMode/ComponentModeDelegateNotificationBus.h

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/Component/ComponentBus.h>
+
+namespace AzFramework
+{
+    //! Provides an interface to notify about ComponentModeDelegate states.
```

Comment:
nit/suggestion: `Provides an interface to notify changes to ComponentModeDelegate state.`